game-core/src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -328,12 +327,18 @@ public void windowClosing(final WindowEvent e) {
     status.setBorder(new EtchedBorder(EtchedBorder.RAISED));
     final JPanel stepPanel = new JPanel();
     stepPanel.setLayout(new GridBagLayout());
-    stepPanel.add(step, new GridBagConstraints(1, 0, 1, 1, 0, 0, GridBagConstraints.EAST, GridBagConstraints.BOTH,
-        new Insets(0, 0, 0, 0), 0, 0));
     stepPanel.add(player, new GridBagConstraints(0, 0, 1, 1, 0, 0, GridBagConstraints.EAST, GridBagConstraints.BOTH,
         new Insets(0, 0, 0, 0), 0, 0));
+    stepPanel.add(step, new GridBagConstraints(1, 0, 1, 1, 0, 0, GridBagConstraints.EAST, GridBagConstraints.BOTH,
+        new Insets(0, 0, 0, 0), 0, 0));
     stepPanel.add(round, new GridBagConstraints(2, 0, 1, 1, 0, 0, GridBagConstraints.EAST, GridBagConstraints.BOTH,
         new Insets(0, 0, 0, 0), 0, 0));
+    if (game.getRandomSource() instanceof PbemDiceRoller) {
```

Comment:
`instanceof` is generally problematic. To avoid that what would you think of perhaps attaching a `isDiceServer()` method to `getRandomSource()`?   so we'd have something like: `getRandomSource().isDiceServer()`

Reply:
I believe doing that would be a lobby incompatible change.